include/verilated_types.h

```diff
@@ -1003,12 +1004,157 @@ std::string VL_TO_STRING(const VlUnpacked<T_Value, T_Depth>& obj) {
     return obj.to_string();
 }
 
+class VlClass;  // See below
+
+//===================================================================
+// Class providing delayed deletion of garbage objects. Objects get deleted only when 'deleteAll()'
+// is called, or the deleter itself is destroyed.
+
+class VlDeleter final {
+    // MEMBERS
+    std::vector<VlClass*> m_garbage;  // Queue of objects to delete
+    VerilatedMutex m_mutex;  // Mutex protecting garbage queue
```

Comment:
Should we instead be tracking deleters per thread?  Seems potentially much faster; large UVM models may be doing thousands of object operations per cycle.

Reply:
Possibly, though maybe let's cross that bridge once we can run UVM? :smiley: I expect we'll need proper garbage collection then as well.